projects/components/src/filter-tabs/filter-tabs.component.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  QueryList,
+  ViewChildren
+} from '@angular/core';
+import { TypedSimpleChanges } from '@hypertrace/common';
+import { FilterTab } from './filter-tab';
+import { FilterTabComponent } from './filter-tab.component';
+
+@Component({
+  selector: 'htc-filter-tabs',
+  styleUrls: ['./filter-tabs.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="filter-tabs" *ngIf="this.activeTab">
+      <div
+        class="active"
+        *ngIf="this.activeElementPosition"
+        [style.left.px]="this.activeElementPosition.left"
+        [style.width.px]="this.activeElementPosition.width"
+      ></div>
+      <div class="container" *ngFor="let tab of this.tabs; let index = index">
+        <div class="divider" *ngIf="index !== 0" [class.adjacent-active]="this.isAdjacentActiveTab(index)"></div>
+        <htc-filter-tab class="tab" [label]="tab.label" (click)="this.setActiveTab(tab)"></htc-filter-tab>
+      </div>
+    </div>
+  `
+})
+export class FilterTabsComponent implements AfterViewInit, OnChanges {
+  @Input()
+  public readonly tabs?: FilterTab[] = [];
+
+  @Input()
+  public activeTab?: FilterTab;
+
+  @Output()
+  public readonly activeTabChange: EventEmitter<FilterTab> = new EventEmitter();
+
+  @ViewChildren(FilterTabComponent, { read: FilterTabComponent })
+  private readonly tabElements!: QueryList<FilterTabComponent>;
+
+  private activeTabIndex?: number;
+  public activeElementPosition?: { left: number; width: number };
+
+  public constructor(private readonly changeDetector: ChangeDetectorRef) {}
+
+  public ngOnChanges(changes: TypedSimpleChanges<this>): void {
+    if (changes.tabs) {
+      this.setActiveTab();
+    }
+  }
+
+  public ngAfterViewInit(): void {
+    this.setActiveTab();
+  }
+
+  public setActiveTab(tab?: FilterTab): void {
+    this.activeTab = tab || (this.tabs && this.tabs[0]);
+    this.activeTabIndex = this.tabs?.indexOf(this.activeTab!);
+    this.activeElementPosition = this.getElementPosition(this.activeTab!, this.activeTabIndex);
+    this.activeTabChange.emit(this.activeTab);
+    this.changeDetector.markForCheck();
```

Comment:
Wouldn't this through View changed after check error?

Reply:
We could probably use QueryListAndChanges

Reply:
Or just make the active tab an observable